From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Thu, 19 Aug 2021 16:28:33 +0200
Subject: launch: board model for RISC-V

On RISC-V there is no default machine type. Invoking QEMU requires to
specify a board model with the -M option. So let's define MACHINE_TYPE
as virt.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
---
 lib/guestfs-internal.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/guestfs-internal.h b/lib/guestfs-internal.h
index ea3924a..5003d8a 100644
--- a/lib/guestfs-internal.h
+++ b/lib/guestfs-internal.h
@@ -148,6 +148,9 @@ guestfs_int_cleanup_gl_recursive_lock_unlock (void *ptr)
 #ifdef __powerpc__
 #define MACHINE_TYPE "pseries"
 #endif
+#ifdef __riscv
+#define MACHINE_TYPE "virt"
+#endif
 
 /* Differences in qemu device names on ARMv7 (virtio-mmio), s/390x
  * (CCW) vs normal hardware with PCI.
